### PR TITLE
Fixes to support e10s Firefox

### LINKED ISSUE #13: moz-spdy-indicator doesn't work with e10s
First version of firefox to have a problem

```
Error: NS_ERROR_UNEXPECTED: Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED) [nsILoadContext.associatedWindow]
Source File: chrome://spdyindicator/content/indicator.jsm
Line: 25
```

### PR DESCRIPTION
**Pull request by bsiegel**
Closes #13 (also likely #18, #19)

### CODE REVIEW

File: chrome/content/indicator.jsm [hunk 1]
```diff
@@ -93,7 +92,19 @@ var SPDYManager = {
         if (!spdyHeader || !spdyHeader.length) return;
 
         // find the browser which this request originated from
-        let domWindow = getLoadContext(subject);
+        let loadContext = getLoadContext(subject);
+        if (!loadContext) return null;
```
Comment (chengsun) [inline chrome/content/indicator.jsm]: Apologies for the nitpick, but this `return null` is inconsistent with the other return statements in this function.

**Comment by chengsun** [comment]
Thanks for doing this. I had a go at trying to support e10s a very long time back, and failed miserably, so I'm very grateful that you managed to pinpoint and fix the problem.

I can't test this right now, but I trust that you have done so. I've given you commit access, so feel free to merge this yourself. I'll look at giving you access on AMO too.

File: chrome/content/indicator.jsm (hunk 1 above)
Comment (bsiegel) [inline chrome/content/indicator.jsm]: Ah quite right! Fixed in 8c80b99

Comment (bsiegel) [inline chrome/content/indicator.jsm]: Ah quite right! Fixed in 8c80b99

**Comment by bsiegel** [comment]
> Thanks for doing this. I had a go at trying to support e10s a very long time back, and failed miserably, so I'm very grateful that you managed to pinpoint and fix the problem.

I'm running it locally here for a while and will keep an eye out for other issues / crashes / errors logged before considering this good to merge. Thanks for the quick reply!

**Comment by chengsun** [comment]
You should be good to go on AMO too. Let me know if you have any issues.

**Comment by bsiegel** [comment]
Awesome, thank you! I'm going to do a little more cleanup here to try to get rid of the unsafe CPOW warnings. Not quite sure how to fix them yet, but will work on it some more later today.

**Comment by chengsun** [comment]
Looks good to me. Merged, and new version sent to AMO.

**Comment by bsiegel** [comment]
Thanks, I hadn't merged this yet because I've been doing some testing and debugging. The changes appear to fix the issue on some tabs but don't seem to work consistently. I've been meaning to drop in to the Mozilla e10s office hours on Tuesdays and see if someone there can pair with me to figure out the issue. These changes also rely on e10s shims which may introduce some performance issues.




On Wed, Apr 27, 2016 at 6:03 PM -0700, "Cheng Sun" <notifications@github.com> wrote:












Looks good to me. Merged, and new version sent to AMO.

**Pull request merged**